src/Adapters/Auth/oauth2.js

```diff
@@ -0,0 +1,231 @@
+ *     "access_token": "an authorized OAuth2 access token for the user",
+ *   }
+ * }
+ */
+
+import logger from '../../logger';
+const Https = require('https');
+const Parse = require('parse/node').Parse;
+const Path = require('path');
+const Url = require('url');
+const Querystring = require('querystring');
+
+const scriptName = Path.basename(__filename, '.js');
+
+// Returns a promise that fulfills if this user id is valid.
+function validateAuthData(authData, options) {
+  const _logger = _getLogger(validateAuthData.name);
+  const loggingEnabled = _isLoggingEnabled(options);
+  if (loggingEnabled) {
+    _logger.verbose('authData = %s, options = %s',  _objectToString(authData), _objectToString(options));
+  }
+  return requestJson(options, authData.access_token, loggingEnabled).then((response) => {
+    if (response && response.active && (!options || !options.hasOwnProperty('useridField') || !options.useridField || authData.id == response[options.useridField])) {
+      return Promise.resolve();
+    }
+    return Promise.reject(new Parse.Error(
+      Parse.Error.OBJECT_NOT_FOUND,
+      'OAuth2 access token is invalid for this user.'));
+  });
+}
+
+function validateAppId(appIds, authData, options) {
+  const _logger = _getLogger(validateAppId.name);
+  const loggingEnabled = _isLoggingEnabled(options);
+  if (loggingEnabled) {
+    _logger.verbose('appIds = %s, authData = %s, options = %s', _objectToString(appIds), _objectToString(authData), _objectToString(options));
+  }
+  if (options && options.hasOwnProperty('appidField') && options.appidField) {
+    if (!appIds.length) {
+      return Promise.reject(new Parse.Error(Parse.Error.OBJECT_NOT_FOUND, 'OAuth2 configuration is missing the client app IDs ("appIds" config parameter).'));
+    }
+    return requestJson(options, authData.access_token, loggingEnabled).then((response) => {
+      const appidField = options.appidField;
+      if (response && response[appidField]) {
+        const responseValue = response[appidField];
+        if (Array.isArray(responseValue)) {
+          for (var idx = responseValue.length - 1; idx >= 0; idx--) {
+            if (appIds.indexOf(responseValue[idx]) != -1) {
+              return Promise.resolve();
+            }
+          }
+        } else {
+          if (appIds.indexOf(responseValue) != -1) {
+            return Promise.resolve();
+          }
+        }
+      }
+      return Promise.reject(new Parse.Error(
+        Parse.Error.OBJECT_NOT_FOUND,
+        'OAuth2: the access_token\'s appID is empty or is not in the list of permitted appIDs in the auth configuration.'));
+    });
+  } else {
+    return Promise.resolve();
+  }
+}
+
+function _isLoggingEnabled(options) {
+  return options && options.debug;
+}
+
+function _objectToString(object) {
+  return JSON.stringify(object, null, 2);
+}
+
+function _getLogger(functionName) {
+  const prefix = scriptName + '.' + functionName + '(): ';
+  var getFormat = function(arg) {
```

Comment:
the logger manages the dates on it's own, please don't add another one.

Reply:
Removed _getLogger() entirely.